File: LeetCode-Java/Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode traverse = head;
        while(head != null)
        {
            while(head.next != null && head.val == head.next.val)
            {
                head.next = head.next.next;
            }
            head = head.next;
        }
        return traverse;
    }
}

File: Leetcode-cpp/remove_duplicates_from_sorted_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* sentinel {new ListNode(0)};
        ListNode* pre = sentinel;
        while (head != nullptr) {
            ListNode* nextElement = head->next;
            while (nextElement != nullptr && nextElement->val == head->val) {
                nextElement = nextElement->next;
            }
            pre->next = head;
            pre = head;
            head->next = nextElement;
            head = nextElement;
        }
        return sentinel->next;
    }
};